lib/FrameManager.js

```diff
@@ -401,22 +401,26 @@ class Frame {
   waitForSelector(selector, options = {}) {
     const timeout = options.timeout || 30000;
     const waitForVisible = !!options.visible;
-    const polling = waitForVisible ? 'raf' : 'mutation';
-    return this.waitForFunction(predicate, {timeout, polling}, selector, waitForVisible);
+    const waitForHidden = !!options.hidden;
+    const polling = waitForVisible || waitForHidden ? 'raf' : 'mutation';
+    return this.waitForFunction(predicate, {timeout, polling}, selector, waitForVisible, waitForHidden);
 
     /**
      * @param {string} selector
      * @param {boolean} waitForVisible
+     * @param {boolean} waitForHidden
      * @return {boolean}
      */
-    function predicate(selector, waitForVisible) {
+    function predicate(selector, waitForVisible, waitForHidden) {
       const node = document.querySelector(selector);
       if (!node)
-        return false;
-      if (!waitForVisible)
+        return waitForHidden;
+      if (!waitForVisible && !waitForHidden)
         return true;
       const style = window.getComputedStyle(node);
-      return style && style.display !== 'none' && style.visibility !== 'hidden';
+      if (waitForVisible)
+        return style && style.display !== 'none' && style.visibility !== 'hidden';
+      return style && (style.display === 'none' || style.visibility === 'hidden');
```

Comment:
Technically the `!style` makes more sense here.

How about:
```js
const style = window.getComputedStyle(node);
const isVisible = style && style.display !== 'none' && style.visibility !== 'hidden';
return waitForVisible === isVisible;
```

Reply:
looks better
``` javascript
return ( waitForVisible === isVisible || waitForHidden === !isVisible);
```